lambda-buffers-codegen/src/LambdaBuffers/Codegen/Rust/Print/LamVal.hs

```diff
@@ -0,0 +1,398 @@
+module LambdaBuffers.Codegen.Rust.Print.LamVal (printValueE, printInstance) where
+
+import Control.Lens ((&), (.~))
+import Control.Monad.Error.Class (MonadError (throwError))
+import Control.Monad.Except (replicateM)
+import Data.Foldable (Foldable (toList))
+import Data.List (sortOn)
+import Data.Map qualified as Map
+import Data.Map.Ordered qualified as OMap
+import Data.ProtoLens (Message (defMessage))
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Traversable (for)
+import LambdaBuffers.Codegen.LamVal qualified as LV
+import LambdaBuffers.Codegen.LamVal.MonadPrint qualified as LV
+import LambdaBuffers.Codegen.Rust.Print.Syntax qualified as R
+import LambdaBuffers.Codegen.Rust.Print.TyDef qualified as TD
+import LambdaBuffers.Compiler.LamTy qualified as LT
+import LambdaBuffers.ProtoCompat qualified as PC
+import Prettyprinter (Doc, Pretty (pretty), align, angles, braces, brackets, colon, comma, dot, dquotes, encloseSep, equals, group, langle, lbracket, line, lparen, parens, pipe, punctuate, rangle, rbracket, rparen, semi, space, vsep, (<+>))
+import Proto.Codegen_Fields qualified as P
+
+caseIntERef :: R.QValName
+caseIntERef = R.Qualified'LibRef (R.MkCrateName "lbr-prelude") (R.MkModuleName "lamval") (R.MkValueName "case_int")
+
+bigInt :: R.QValName
+bigInt = R.Qualified'LibRef (R.MkCrateName "num-bigint") (R.MkModuleName "") (R.MkValueName "BigInt")
+
+vecAsSlice :: R.QValName
+vecAsSlice = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "vec") (R.MkValueName "Vec::as_slice")
+
+vecMacro :: R.QValName
+vecMacro = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "") (R.MkValueName "vec!")
+
+fromU32Trait :: R.QTraitName
+fromU32Trait = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "convert") (R.MkTraitName "From<u32>")
+
+fromStrTrait :: R.QTraitName
+fromStrTrait = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "convert") (R.MkTraitName "From<&str>")
+
+cloneTrait :: R.QTraitName
+cloneTrait = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "clone") (R.MkTraitName "Clone")
+
+boxNew :: R.QValName
+boxNew = R.Qualified'LibRef (R.MkCrateName "std") (R.MkModuleName "boxed") (R.MkValueName "Box::new")
+
+phantomData :: R.QTyName
+phantomData = R.qLibRef R.MkTyName "std" "marker" "PhantomData"
```

Comment:
I would also consider grouping this under Refs.hs

Reply:
I should've done that